src/components/CodeSnippet/CodeSnippet.tsx

```diff
@@ -0,0 +1,145 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+import type { HTMLProps } from "react";
+
+export type Props = {
+  className?: string;
+  blocks: Array<CodeSnippetBlockProps>;
+};
+
+export default function CodeSnippet({ className, blocks }: Props): JSX.Element {
+  return (
+    <div className={classNames("p-code-snippet", className)}>
+      {blocks.map((props, i) => (
+        <CodeSnippetBlock
+          key={`code-snippet-block-${i}`}
+          {...props}
+        ></CodeSnippetBlock>
+      ))}
+    </div>
+  );
+}
+
+CodeSnippet.propTypes = {
+  blocks: PropTypes.array.isRequired,
+  className: PropTypes.string,
+};
+
+type CodeSnippetBlockProps = {
+  code: string;
+  title?: string;
+  appearance?: "numbered" | "linuxPrompt" | "windowsPrompt" | "url";
```

Comment:
These specific strings get used throughout the component, so instead of have to type the exact string you could use an enum here:

```typescript
export enum CodeSnippetAppearance {
  LINUX_PROMPT = "linuxPrompt",
  NUMBERED = "numbered",
  URL = "url",
  WINDOWS_PROMPT = "windowsPrompt",
}
```

Appearance would be defined as `appearance?: CodeSnippetAppearance`.

Below you would then do things like `appearance === CodeSnippetAppearance.LINUX_PROMPT`

Reply:
Right, I thought about enums, but when I looked for example in react-components repo I've seen list list of strings used. But that's a good suggestion. And now I see that Icons use enums as well.